gnuradio-runtime/lib/buffer_context.cc

```diff
@@ -0,0 +1,32 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2015 Free Software Foundation, Inc.
+ *
+ * This file is part of GNU Radio
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ */
+#include <gnuradio/buffer_context.h>
+
+namespace gr {
+
+std::ostream& operator<<(std::ostream& os, const buffer_context& context)
+{
+    switch (context) {
+    case buffer_context::DEFAULT_INVALID:
```

Comment:
Is this list of contexts generic enough for all types of buffers?  Is ingress/egress a sufficient set of cases that buffers would need to handle?

Reply:
That's a very good question. I think so, but let's discuss if you have other ideas.